### PR TITLE
getting started examples fix and improvements 9/23/2023

### PR DESCRIPTION
**Pull request by talip**
- protos submodule to point to latest github.com/vectara/protos
- fix and cleanup for java and python examples

### CODE REVIEW

**Comment by talip** [comment]
@amin2718 can you preview review this?

File: language-examples/java/rest/README.md [hunk 1]
```diff
@@ -17,16 +17,22 @@ There are two supported authentication methods Vectara.
...
-   authentication library and will make it available for subsequent projects.
-3. cd to `vectara-java-demo/java/rest` directory.
-4. run `mvn package` (This will create a target directory with a shaded JAR)
-5. Run the jar file with a command like following:
+2. Build the authentication library if you haven't already.
+   ```shell
+   cd language-examples/java/auth
+   mvn install
+   ```
+3. Build the rest example. This will create a target directory with a shaded JAR.
```
Comment (eskibars) [inline language-examples/java/rest/README.md:30]: s/rest/REST/

File: language-examples/java/rest/src/main/java/com/vectara/examples/rest/RestArgs.java [hunk 2]
```diff
@@ -52,4 +52,9 @@ public class RestArgs {
       description = "API key retrieved from Vectara console")
   @Nullable
   String apiKey = null;
+  
+  @Parameter(
+      names= {"--query"},
+      description = "The query you want to send to the backend.")
```
Comment (eskibars) [inline language-examples/java/rest/src/main/java/com/vectara/examples/rest/RestArgs.java:58]: I think "the backend" here sounds a bit weird.  I think "to Vectara" sounds more natural

**Review by eskibars — CHANGES_REQUESTED** [review]
Mostly LGTM.  I'm not really in a place to comment on the Java code myself, but left a few pretty minor comments on the strings/markdown

File: language-examples/java/rest/README.md (hunk 1 above)
Comment (talip) [reply language-examples/java/rest/README.md:30]: fixed.

File: language-examples/java/rest/src/main/java/com/vectara/examples/rest/RestArgs.java (hunk 2 above)
Comment (talip) [reply language-examples/java/rest/src/main/java/com/vectara/examples/rest/RestArgs.java]: fixed.

File: language-examples/java/auth/src/main/java/com/vectara/auth/JwtFetcher.java [hunk 3]
```diff
@@ -102,7 +101,7 @@ public String fetchClientCredentialsJwt() {
         return null;
       }
       if (map.containsKey("access_token")) {
-        return JWT.decode(String.valueOf(map.get("access_token"))).getToken();
+        return (String) map.get("access_token");
```
Comment (amin2718) [inline language-examples/java/auth/src/main/java/com/vectara/auth/JwtFetcher.java:104]: It's more robust to just say:
```
String.valueOf(map.get("access_token"));
```
This code cannot crash for any reason.

**Review by amin2718 — APPROVED** [review]
Some of these changes look like ones that I made (the additional --query parameter, for example). Why did you have to add those back in again?

File: language-examples/java/auth/src/main/java/com/vectara/auth/JwtFetcher.java (hunk 3 above)
Comment (talip) [reply language-examples/java/auth/src/main/java/com/vectara/auth/JwtFetcher.java]: done

**Review by eskibars — APPROVED** [review]
LGTM, thanks

**Pull request merged**